A41-xds-rbac.md

```diff
@@ -62,107 +62,134 @@ Supporting the RBAC HTTP filter on server-side leverages [A36: xDS-Enabled
 
 The [RBAC policy][] has many fields. All current (Envoy-implemented) fields will
 be considered in gRPC. However, some fields may have pre-determined values or
 behavior. At this time, if the `RBAC.action` is `Action.LOG` then the policy
 will be completely ignored, as if RBAC was not configurated. CEL fields are not
-supported, so if `Policy.condition` or `Policy.checked_condition` is present the
-XdsClient must NACK the configuration.
+supported, so `Policy.condition` and `Policy.checked_condition` must cause a
+validation failure if present. It is also a validation failure if Permission or
+Principal has a `header` matcher for a `grpc-`-prefixed header name or
+`:scheme`. As described in A39, validation failures for filter configuration
+causes the listener to be NACKed.
 
 The following fields of `Permission` may not be obvious how they map to gRPC:
 
 | Permission Field | gRPC Equivalent |
 | ---------------- | --------------- |
-| header           | Metadata (with caveats) |
-| url_path         | Fully-qualified RPC method name with leading slash |
+| header           | Metadata (with caveats below) |
+| url_path         | Fully-qualified RPC method name with leading slash. Same as `:path` header |
 | destination_ip   | Local address for this connection |
 | destination_port | Local port for this connection |
 | metadata         | Hard-coded as empty; never matches |
 | requested_server_name | Hard-coded as empty string |
 
-Be strongly aware that Envoy Metadata has no relation to gRPC's Metadata.
 Because matching supports NOT, the matcher must still be processed even if a
 rule contains references to things that don't generally match; it is not trivial
 to "optimize out" the never-matching rules.
 
-The `header` field is not entirely 1:1 with gRPC Metadata. To begin with, gRPC
-Metadata isn't 100% consistent cross-language in its handling of hop-to-hop
-headers (e.g., `TE`) and pseudo-headers. For this design, `headers` can include
-`:method`, `:scheme`, `:authority`, and`:path` matchers and they must match the
-values received on-the-wire independent of whether they are stored in Metadata
-or in separate APIs. It is unspecified whether hop-to-hop headers are matched.
+The `header` field is not entirely 1:1 with gRPC Metadata, in part because which
+HTTP headers are present in Metadata is not 100% consistent cross-language.
+For this design, `headers` can include `:method`, `:authority`, and`:path`
+matchers and they should match the values received on-the-wire independent of
+whether they are stored in Metadata or in separate APIs. `:method` can be
+hard-coded to `POST` if unavailable and a code audit confirms the server denies
+requests for all other method types. Implementations must not match [hop-by-hop
+headers][]. Since hop-by-hop headers [are not used in HTTP/2 except for `te:
+trailers`][rfc7540 connection header], implementations must ensure they are
+disallowing the `Connection` header and not matching the `TE` header.
```

Comment:
So the intent here that if a rule matches on the `te` header, it will not match, but if that is a `not` rule, it *will* match?  That doesn't seem obvious from the wording.  How about saying something like this:

"""
If a rule attempts to match against the `te` header, the RBAC implementation must treat that header as non-existing, regardless of what was actually received on the wire.
"""